lighthouse-extension/app/src/popup.js

```diff
@@ -182,15 +146,58 @@ document.addEventListener('DOMContentLoaded', _ => {
         feedbackEl.appendChild(buildReportErrorLink(err));
       }
 
-      stopSpinner();
+      hideRunningSubpage();
       background.console.error(err);
     });
   });
+}
+
+/**
+ * Generates a document fragment containing a list of checkboxes and labels
+ * for the aggregation categories.
+ * @param {!Object<boolean>} selectedAggregations
+ * @param {!Window} background Reference to the extension's background page.
+ * @return {!DocumentFragment}
+ */
+function generateOptionsList(list, selectedAggregations, background) {
+  const frag = document.createDocumentFragment();
+
+  const defaultAggregations = background.getDefaultAggregations();
+  defaultAggregations.forEach(aggregation => {
+    const isChecked = selectedAggregations[aggregation.name];
+    frag.appendChild(createOptionItem(aggregation.name, isChecked));
+  });
+
+  return frag;
+}
```

Comment:
(could still just pass in `background`, of course, since the caller already has a reference to it)